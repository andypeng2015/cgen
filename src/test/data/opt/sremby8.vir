module foo

export function w $foo(w %x) {
@start:
  %x = rem.w %x, 8_w
  ret %x
}
