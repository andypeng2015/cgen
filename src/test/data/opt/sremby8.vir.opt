module foo

export function w $foo(w %x) {
@2:
  %5 = asr.w %x, 0x1f_w ; @9
  %6 = lsr.w %5, 0x1d_w ; @10
  %7 = add.w %x, %6 ; @11
  %8 = and.w %7, 0x7_w ; @12
  %9 = sub.w %8, %6 ; @13
  ret %9
}
