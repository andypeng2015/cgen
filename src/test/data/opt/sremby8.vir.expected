module foo

export function w $foo(w %x) {
@0:
  %0 = asr.w %x, 0x1f_w ; @2
  %1 = lsr.w %0, 0x1d_w ; @3
  %2 = add.w %x, %1 ; @4
  %3 = and.w %2, 0x7_w ; @5
  %4 = sub.w %3, %1 ; @6
  ret %4
}
